### PR TITLE
Change output and order of deps installation

### PR DESCRIPTION
**Pull request by mpeterv**
This PR makes output for `luarocks install` and other commands that satisfy dependencies closer to what #496 does:

* Process dependencies in the order of `rockspec.dependencies` array.
* Announce each missing dependency just before installing it, instead of listing them all beforehand (makes it easier to understand why each rock is installed in a long list).
* Show what versions are already installed for missing dependencies.
* Remove not very useful (IMO) lines like `switching to build mode` and `updating manifest`.

See comparison of outputs of `luarocks install sailor`: https://gist.github.com/mpeterv/a43fbcdeadaec40dd6fa3284ea7f07dd

In a nutshell:

```
Using https://luarocks.org/valua-0.3-1.src.rock... switching to 'build' mode
Updating manifest for /home/mpeterv/luarocks/here/lib/luarocks/rocks
valua 0.3-1 is now built and installed in /home/mpeterv/luarocks/here (license: MIT)
```

changes to

```
sailor 0.5-4 depends on valua >= 0.2.2 (missing)
Installing https://luarocks.org/valua-0.3-1.src.rock
valua 0.3-1 is now installed in /home/mpeterv/luarocks/here (license: MIT)
```

### CODE REVIEW

**Comment by hishamhm** [comment]
The new output looks good, but can we still have the list of missing dependencies displayed at the top? I think it gives a good preliminary indication when "there's a lot to do".

**Comment by mpeterv** [comment]
Readded the list. Also replaced `missing` label with `not installed` to avoid repetition in the list header.

**Comment by hishamhm** [comment]
Thank you!
